frontend-react/src/views/Drives/components/DrivesList/index.js

```diff
@@ -0,0 +1,168 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Accordion from '@material-ui/core/Accordion';
+import AccordionSummary from '@material-ui/core/AccordionSummary';
+import AccordionDetails from '@material-ui/core/AccordionDetails';
+import Typography from '@material-ui/core/Typography';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Pagination from '@material-ui/lab/Pagination';
+import { mockDrives } from './mockItems';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    minWidth: '80%',
+    maxWidth: '1100px',
+    marginLeft: 'auto',
+    marginRight: 'auto',
+  },
+  accordion: {
+    margin: '3px',
+    borderRadius: '10px',
+    '&:before': {
+      backgroundColor: 'transparent !important',
+    },
+  },
+  heading: {
+    fontSize: theme.typography.pxToRem(15),
+    fontWeight: theme.typography.fontWeightRegular,
+  },
+  verified: {
+    background: 'linear-gradient(90deg, #28a745 0.5%, #FAF9F6 0%)',
+    borderRadius: '10px',
+  },
+  unverified: {
+    background: 'linear-gradient(90deg, #ffc107 0.5%, #FAF9F6 0%)',
+    borderRadius: '10px',
+  },
+  pagination: {
+    display: 'flex',
+    justifyContent: 'center',
+    marginTop: '20px',
+  },
+}));
+
+const totalMilage = () => {
+  let milage = 0;
+
+  mockDrives.map((drive) => {
```

Comment:
Array.prototype.map() is a method that creates a new array populated with the results of calling a provided callback function on every element in the calling array. To achieve the effect you expect, you should use Array.prototype.reduce() method.

Reply:
Rewritten to utilized reduce and use Memo